include/cetl/unbounded_variant.hpp

```diff
@@ -48,36 +48,463 @@ class bad_unbounded_variant_access : public std::bad_cast
 #endif  // defined(__cpp_exceptions) || defined(CETL_DOXYGEN)
 
 // Forward declarations
-template <std::size_t Footprint, bool Copyable, bool Movable, std::size_t Alignment>
+template <std::size_t Footprint, bool Copyable, bool Movable, std::size_t Alignment, bool IsPmr>
 class unbounded_variant;
 
 namespace detail
 {
 
+inline std::nullptr_t throw_bad_alloc()
+{
+#if defined(__cpp_exceptions)
+    throw std::bad_alloc();
+#else
+    return nullptr;
+#endif
+}
+
+[[noreturn]] inline void throw_bad_unbounded_variant_access()
+{
+#if defined(__cpp_exceptions)
+    throw bad_unbounded_variant_access();
+#else
+    CETL_DEBUG_ASSERT(false, "bad_unbounded_variant_access");
+    std::terminate();
+#endif
+}
+
+// MARK: - Storage policy.
+//
+template <std::size_t Footprint, bool IsPmr, std::size_t Alignment>
+struct base_storage;
+//
+// In-Place only case.
+template <std::size_t Footprint, std::size_t Alignment>
+struct base_storage<Footprint, false /*IsPmr*/, Alignment>
+{
+    /// @brief Allocates enough raw storage for a target value.
+    ///
+    /// @param size_bytes Size of the target type in bytes.
+    /// @return Pointer to the storage (never `nullptr` for this specialization).
+    ///
+    CETL_NODISCARD void* alloc_new_raw_storage(const std::size_t size_bytes) noexcept
+    {
+        CETL_DEBUG_ASSERT(0UL < size_bytes, "");
+        CETL_DEBUG_ASSERT(size_bytes <= Footprint, "");
+        CETL_DEBUG_ASSERT(0UL == value_size_, "This object is expected to be a brand new one.");
+
+        // We need to store (presumably) allocated size b/c this is important
+        // for detection of the "valueless by exception" state in case something goes wrong later
+        // (like in-place value construction failure) - in such failure state
+        // there will `>0` value size BUT `nullptr` handlers (f.e. `value_destroyer_`).
+        //
+        value_size_ = size_bytes;
+
+        return inplace_buffer_;
```

Comment:
should we not check that size_bytes is <= Footprint and return null if not?

Reply:
No, for non-pmr there is no such thing. Please see static assert at `make_handlers` - this method is one of the core entry points when it come to creation of a new unbounded variable. So it won't compile (and there is compile test to ensure this, see `test_unbounded_variant_footprint_set.cpp`).

There are also two other static asserts on attempt to read an arbitrary value (see `get_ptr` overloads, and corresponding `test_unbounded_variant_footprint_get_[non_]const.cpp` compile tests).